lib/plugins/inspec-habitat/README.md

```diff
@@ -0,0 +1,150 @@
+# InSpec Habitat Plugin
+
+## Summary
+
+This plugin allows you to do the following:
+  1. Add Habitat configuration to a profile
+  2. Create/Upload a Habitat package from an InSpec profile
+
+Creating a [Habitat](https://www.habitat.sh/) package from an InSpec profile
+allows you to execute that profile as a service (via a Habitat Supervisor) on
+any Linux based platform.
+
+When running as a service, an InSpec report will be created in JSON format (by
+default at `/hab/svc/YOUR_SERVICE/logs/inspec_last_run.json`).  Additionally, a
+log of the last run will be located at
+`/hab/svc/YOUR_SERVICE/logs/inspec_log.txt` and CLI output is viewable in
+the Supervisor logs by default. You can also configure this service to report
+to [Chef Automate](https://www.chef.io/automate/).
+
+See below for usage instructions.
+
+## Plugin Usage
+
+### Adding Habitat Configuration to an InSpec Profile
+
+Run the following command:
+
+```
+inspec habitat profile setup PATH
+```
+
+This will create the following files:
+  - habitat/plan.sh (Provides build time instructions to Habitat)
+  - habitat/default.toml (Used to configure the running Habitat service)
+  - habitat/hooks/run (Shell script to execute this profile as a service)
+  - habitat/config/inspec_exec_config.json (JSON for `inspec exec` CLI options)
+
+### Creating a Habitat Package
+
+> This command requires Habitat to be installed and configured. For instructions
+on how to do that see [here](https://www.habitat.sh/docs/install-habitat/).
+
+Run the following command:
+
+```
+inspec habitat profile create PATH
```

Comment:
With no ENV variables set, this still throws a stacktrace:

```
[cwolfe@lodi inspec-review]$ tree my-hab-profile/
my-hab-profile/
├── README.md
├── controls
│   └── example.rb
├── habitat
│   ├── config
│   │   └── inspec_exec_config.json
│   ├── default.toml
│   ├── hooks
│   │   └── run
│   └── plan.sh
├── inspec.yml
└── libraries

5 directories, 7 files
[cwolfe@lodi inspec-review]$ be inspec habitat profile create my-hab-profile
[2019-02-28T09:16:19-05:00] ERROR: Unable to determine Habitat origin name.
[2019-02-28T09:16:19-05:00] ERROR: Run `hab setup` or set the HAB_ORIGIN environment variable.
[2019-02-28T09:16:19-05:00] ERROR: Unable to create Habitat artifact.
bundler: failed to load command: inspec (/Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/bin/inspec)
RuntimeError:
  /Users/cwolfe/sandbox/inspec/inspec-review/lib/plugins/inspec-habitat/lib/inspec-habitat/profile.rb:266:in `exit_with_error'
  /Users/cwolfe/sandbox/inspec/inspec-review/lib/plugins/inspec-habitat/lib/inspec-habitat/profile.rb:179:in `verify_habitat_setup'
  /Users/cwolfe/sandbox/inspec/inspec-review/lib/plugins/inspec-habitat/lib/inspec-habitat/profile.rb:30:in `create'
  /Users/cwolfe/sandbox/inspec/inspec-review/lib/plugins/inspec-habitat/lib/inspec-habitat/cli.rb:17:in `create'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:115:in `invoke'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor.rb:238:in `block in subcommand'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:115:in `invoke'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor.rb:238:in `block in subcommand'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
  /Users/cwolfe/sandbox/inspec/inspec-review/bin/inspec:12:in `<top (required)>'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/bin/inspec:23:in `load'
  /Users/cwolfe/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/bin/inspec:23:in `<top (required)>'
[cwolfe@lodi inspec-review]$ git log --oneline -n1; date
b533958a (HEAD -> ja/rewrite-habitat-plugin, origin/ja/rewrite-habitat-plugin) Remove stacktrace being sent to `exit_with_error`
Thu Feb 28 09:18:17 EST 2019
[cwolfe@lodi inspec-review]$
```

Reply:
That stack trace is coming from Bundler. Not the InSpec process.

I'm not sure I can suppress that. Unless you know of a way?

Reply:
Nope, I'm an idiot! I called `raise` without handling any exceptions. You get a different stack trace when not using Bundler, but a stack trace is a stack trace.